src/GeneratorV2/Process/Overloaders/VectorOverloader.cs

```diff
@@ -0,0 +1,188 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using GeneratorV2.Data;
+using GeneratorV2.Writing;
+
+namespace GeneratorV2.Process.Overloaders
+{
+    public class VectorOverloader : IOverloader
+    {
+        public bool TryGenerateOverloads(Overload overload, out List<Overload>? newOverloads)
+        {
+            string name = overload.OverloadName;
+
+            // First the name is passed so we can check for the correct vector type.
+            int vIndex = name.IndexOf('v');
+            if (vIndex == -1)
+            {
+                newOverloads = null;
+                return false;
+            }
+            char typeChar = name[vIndex - 1];
+            string baseType;
+            switch (typeChar)
+            {
+                case 'i':
+                    baseType = "int";
+                    break;
+                case'd':
+                    baseType = "double";
+                    break;
+                case 'f':
+                    baseType = "float";
+                    break;
+                case 'h':
+                    baseType = "half";
+                    break;
+                default:
+                    // Logger.Warning($"Couldn't turn {typeChar} into a vector type.");
+                    newOverloads = null;
+                    return false;
+            }
+
+            bool foundVector = false;
+            NameTable nameTable = overload.NameTable.New();
+            Parameter[] parameters = overload.InputParameters;
+            List<Parameter> newParameters = parameters.ToList();
+            List<Parameter> pointerParameters = new List<Parameter>();
+            List<Parameter> vectorParameters = new List<Parameter>();
+            List<Parameter> countParameters = new List<Parameter>();
+            for (int i = 0; i < parameters.Length; i++)
+            {
+                Parameter parameter = parameters[i];
+                if (parameter.Type is CSPointer {BaseType: CSType csType} &&
```

Comment:
Nitpick: spaces between the `{` and the `}`
```cs
{ BaseType: CSType csType }
```

Reply:
This could also be done with:
```cs
parameter.Type is CSPointer pointer && pointer.BaseType is CSType csType && csType.TypeName == baseType
```
this might be easier to read as it doesn't require as much knowledge about C# pattern matching.
Personally I'm fine with this like it is.